force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -36,17 +36,29 @@ public with sharing class DummyJSONCallout {
 
 	public static void getDummyJSONUserFromId(String dummyUserId) {
 		// Create HTTP request to send.
-
+        HttpRequest request = new HttpRequest();
 		// Set the endpoint URL. Use direct URL or for best practices use Named Credential.
-
+        request.setEndpoint('https://dummyjson.com/user/' + dummyUserId);
 		// Set the HTTP method to GET.
+        request.setMethod('GET');
 
 		// Send the HTTP request and get the response.
+		Http  http = new Http();
+		HttpResponse response = Http.send(Request);
+		if (response.getStatusCode() == 200) {
 
 		// If the HTTP response code is successful, parse the JSON response and update the contact.
-		Contact cont = null;
-		upsert cont DummyJSON_Id__c; //insert/update from the JSON response using the external id (dummyUserId)
+		Contact cont = createContactFromJson(response.getBody());
+		upsert cont DummyJSON_Id__c ; //insert/update from the JSON response using the external id (dummyUserId)
 	}
+}
+
+    @future (callout = true)
+	public static void getDummyJSONUserFromIdFuture(String dummyUserId) {
```

Comment:
Good using another method to create an @future option so you maintain the option for both synchronous and asynchronous.